pygments/lexers/jsonnet.py

```diff
@@ -0,0 +1,175 @@
+"""
+    pygments.lexers.jsonnet
+    ~~~~~~~~~~~~~~~~~~~~~~~
+
+    Lexer for Jsonnet data templating language.
+
+    :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+from pygments.lexer import (
+    include,
+    RegexLexer,
+    words,
+)
+from pygments.token import (
+    Comment,
+    Keyword,
+    Name,
+    Number,
+    Operator,
+    Punctuation,
+    String,
+    Whitespace,
+)
+
+__all__ = ['JsonnetLexer']
+
+jsonnet_token = r'[^\W\d]\w*'
+jsonnet_function_token = jsonnet_token + r'(?=\()'
+
+
+def string_rules(quote_mark):
+    return [
+        (r"[^{}\\]".format(quote_mark), String),
+        (r"\\.", String.Escape),
+        (quote_mark, String, '#pop'),
+    ]
+
+
+def quoted_field_name(quote_mark):
+    return [
+        (r'([^{quote}\\]|\\.)*{quote}'.format(quote=quote_mark),
+         Name.Variable, 'field_separator')
+    ]
+
+
+class JsonnetLexer(RegexLexer):
+    name = 'Jsonnet'
+    aliases = ['jsonnet']
+    filenames = ['*.jsonnet', '*.libsonnet']
+    tokens = {
+        # Not used by itself
+        '_comments': [
+            (r'(//|#).*\n', Comment.Single),
+            (r'/\*\*([^/]|/(?!\*))*\*/', String.Doc),
+            (r'/\*([^/]|/(?!\*))*\*/', Comment),
+        ],
+        'root': [
+            include('_comments'),
+            (r"@'.*'", String),
+            (r'@".*"', String),
+            (r"'", String, 'singlestring'),
+            (r'"', String, 'doublestring'),
+            (r'(?s:\|\|\|.*\|\|\|)', String),
+            (r'[+-]?[0-9]+(.[0-9])?', Number.Float),
```

Comment:
Jsonnet only has one number type, which is an IEEE754 64-bit floating point number.  I thought that it made sense for any numeric literal to produce the same type in the highlighter, since it produces the same type when parsed.  Was that the wrong approach?

Reply:
Done